### ISSUE TITLE
printViewHierarchy prints over 12000 lines of the same text

### ISSUE THREAD

**Issue by paulz** [kif-framework/KIF#1146]
When using `KIF_PRINTVIEWTREEONFAILURE`
and the test fails on a view with UIDatePicker I am getting failure log with over 12000 lines containing:

`UITableViewCellAccessibilityElement, label: (null), traits: none`

and output looks like:
```
UIWindow
|	UITransitionView
|	|	UIDropShadowView
|	|	|	UILayoutContainerView
|	|	|	|	UINavigationTransitionView
|	|	|	|	|	UIViewControllerWrapperView
|	|	|	|	|	|	UITableView, identifier: reminders table
|	|	|	|	|	|	UITableViewCellAccessibilityElement, label: (null), traits: none
|	|	|	|	|	|	UITableViewCellAccessibilityElement, label: (null), traits: none
|	|	|	|	|	|	UITableViewCellAccessibilityElement, label: (null), traits: none
|	|	|	|	|	|	UITableViewCellAccessibilityElement, label: (null), traits: none
|	|	|	|	|	|	UITableViewCellAccessibilityElement, label: (null), traits: none
|	|	|	|	|	|	|	UITableViewCell (invisible)
|	|	|	|	|	|	|	|	UITableViewCellContentView
|	|	|	|	|	|	|	|	|	UIDatePicker (enabled) (not selected) (not highlighted)
|	|	|	|	|	|	|	|	|	|	_UIDatePickerView
|	|	|	|	|	|	|	|	|	|	UIAccessibilityPickerComponent, label: (null), value: 11 o’clock, traits: adjustable
|	|	|	|	|	|	|	|	|	|	UIAccessibilityPickerComponent, label: (null), value: 25 minutes, traits: adjustable
|	|	|	|	|	|	|	|	|	|	UIAccessibilityPickerComponent, label: (null), value: AM, traits: adjustable
|	|	|	|	|	|	|	|	|	|	|	UIView
|	|	|	|	|	|	|	|	|	|	|	|	UIPickerColumnView
|	|	|	|	|	|	|	|	|	|	|	|	|	UIView
|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIPickerTableView
|	|	|	|	|	|	|	|	|	|	|	|	|	|	UITableViewCellAccessibilityElement, label: (null), traits: none
|	|	|	|	|	|	|	|	|	|	|	|	|	|	UITableViewCellAccessibilityElement, label: (null), traits: none
|	|	|	|	|	|	|	|	|	|	|	|	|	|	UITableViewCellAccessibilityElement, label: (null), traits: none
|	|	|	|	|	|	|	|	|	|	|	|	|	|	UITableViewCellAccessibilityElement, label: (null), traits: none
|	|	|	|	|	|	|	|	|	|	|	|	|	|	UITableViewCellAccessibilityElement, label: (null), traits: none

+ over 12000 times

|	|	|	|	|	|	|	|	|	|	|	|	|	|	UITableViewCellAccessibilityElement, label: (null), traits: none
|	|	|	|	|	|	|	|	|	|	|	|	|	|	UITableViewCellAccessibilityElement, label: (null), traits: none
|	|	|	|	|	|	|	|	|	|	|	|	|	|	UITableViewCellAccessibilityElement, label: (null), traits: none
|	|	|	|	|	|	|	|	|	|	|	|	|	|	UITableViewCellAccessibilityElement, label: (null), traits: none
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIPickerTableViewWrapperCell
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIAccessibilityElementMockView, label: 00 minutes, traits: none
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UITableViewCellContentView
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIView
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIDatePickerContentView, label: 00 minutes
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UILabel, label: 00
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIPickerTableViewWrapperCell
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIAccessibilityElementMockView, label: 05 minutes, traits: none
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UITableViewCellContentView
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIView
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIDatePickerContentView, label: 05 minutes
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UILabel, label: 05
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIPickerTableViewWrapperCell
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIAccessibilityElementMockView, label: 10 minutes, traits: none
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UITableViewCellContentView
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIView
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIDatePickerContentView, label: 10 minutes
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UILabel, label: 10
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	_UIScrollViewScrollIndicator, label: Horizontal scroll bar, 1 page
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIView
|	|	|	|	|	|	|	|	|	|	|	|	UIPickerColumnView
|	|	|	|	|	|	|	|	|	|	|	|	|	UIView
|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIPickerTableView
|	|	|	|	|	|	|	|	|	|	|	|	|	|	UITableViewCellAccessibilityElement, label: (null), traits: none
|	|	|	|	|	|	|	|	|	|	|	|	|	|	UITableViewCellAccessibilityElement, label: (null), traits: none
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIPickerTableViewWrapperCell
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIAccessibilityElementMockView, label: PM, traits: none
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UITableViewCellContentView
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIView
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIDatePickerContentView, label: PM
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UILabel, label: PM
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIPickerTableViewWrapperCell
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIAccessibilityElementMockView, label: AM, traits: none
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UITableViewCellContentView
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIView
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIDatePickerContentView, label: AM
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UILabel, label: AM
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	_UIScrollViewScrollIndicator, label: Horizontal scroll bar, 1 page
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIView
|	|	|	|	|	|	|	|	|	|	|	|	|	UIView
|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIPickerTableView
|	|	|	|	|	|	|	|	|	|	|	|	|	|	UITableViewCellAccessibilityElement, label: (null), traits: none
|	|	|	|	|	|	|	|	|	|	|	|	|	|	UITableViewCellAccessibilityElement, label: (null), traits: none
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIPickerTableViewWrapperCell
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIAccessibilityElementMockView, label: PM, traits: none
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UITableViewCellContentView
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIView
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIDatePickerContentView, label: PM
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UILabel, label: PM
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	_UIScrollViewScrollIndicator, label: Horizontal scroll bar, 1 page
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIView
|	|	|	|	|	|	|	|	|	|	|	|	|	UIView
|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIPickerTableView
|	|	|	|	|	|	|	|	|	|	|	|	|	|	UITableViewCellAccessibilityElement, label: (null), traits: none
|	|	|	|	|	|	|	|	|	|	|	|	|	|	UITableViewCellAccessibilityElement, label: (null), traits: none
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIPickerTableViewWrapperCell
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIAccessibilityElementMockView, label: PM, traits: none
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UITableViewCellContentView
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIView
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIDatePickerContentView, label: PM
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UILabel, label: PM
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIPickerTableViewWrapperCell
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIAccessibilityElementMockView, label: AM, traits: none
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UITableViewCellContentView
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIView
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIDatePickerContentView, label: AM
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UILabel, label: AM
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	_UIScrollViewScrollIndicator, label: Horizontal scroll bar, 1 page
|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	|	UIView
|	|	|	|	|	|	|	|	|	|	|	UIView
|	|	|	|	|	|	|	|	|	|	|	UIView
|	|	|	|	|	|	|	|	_UITableViewCellSeparatorView
|	|	|	|	|	|	|	|	_UITableViewCellSeparatorView
```

**Comment by justinseanmartin** [comment]
It seems reasonable to special case the output when walking into a UIDatePicker in the `-[UIView(Debugging) printViewHierarchyWithIndentation:]` method. It already has some special casing for `UIImageView` and `UIControl`. I think we'd probably want to replace the 12k lines with a single line with the properties of the `UIDatePicker` that control (at least date, min, max, mode?, minuteInterval?).

Is this something you'd be willing to contribute? I'd be happy to review and merge the change if so.

**Comment by dostrander** [comment]
Fixed in PR  #1157
